Add ability to search for method/field usage and use custom update sites (#14)
jenkins-infra/usage-in-plugins: USAGE.adoc

Writing output/usage-by-api.json
Writing output/usage-by-api.html
----

=== Finding Usage of Methods and Fields

The command line flag `-M filename.txt` or `--additionalMethods filename.txt` allows specific methods to be searched for.

> The additional flag `-i` or `--onlyIncludeSpecified` will filter the output report to only show usage of the methods specified.

Reviewer: nit:
```suggestion
Use the additional flag `-i` or `--onlyIncludeSpecified` to only show usage of the methods specified.
```

> For example, to find usage of `hudson.util.Secret.toString`, the following line should be included in the specified file:

Reviewer: nit:
```suggestion
For example, to find usage of `hudson.util.Secret.toString`, include the following line in the specified file:
```

[source]
----
hudson/util/Secret#toString
----

> Class names should be specified in their fully qualified form (i.e., inner classes should be referenced via `Outer$Inner` rather than by their canonical names).

Reviewer: More nit:
```suggestion
Specify class names in their fully qualified form (i.e., inner classes should be referenced via `Outer$Inner` rather than by their canonical names).
```

> Method names should be specified after the hash.

Reviewer: ```suggestion
Specify method names after the hash.
```

> Note that you cannot specify which method overload signature, but they are included in reports.

Reviewer: Not sure I'm getting this one right.
```suggestion
You cannot specify distinct method signatures. All variants are included in reports.
```

> The command line flag `-F filename.txt` or `--additionalFields filename.txt` allows specific field usages to be found.

Reviewer: ```suggestion
The command line flag `-F filename.txt` or `--additionalFields filename.txt` allows you to find specific field usages.
```

The syntax here is the same as for specifying method names.

> The additional flag `-i` will filter the output report to only show specified fields.

Reviewer: Actually, this needs to be pulled out and discussed as it applies to all operations generally and not repeated for each different type.
